docs/reference/module-api/module-overview.md

```diff
@@ -257,14 +257,231 @@ modules.export = {
+
+If a piece type included the `insert` in its `extendMethods` section to alter the piece titles, it might look like this:
+
+```javascript
+  extendMethods(self) {
+    return {
+      insert(_super, req, piece, options) {
+        piece.title = `🆕 ${piece.title}`;
+
+        _super(req, piece, options);
+      }
+    }
+  }
+```
+
+::: warning
+Extension functions should *always* use the `_super` argument to call the original function. If you want to *completely* overwrite the inherited function, instead add a matching function in the section without the "extend" prefix. For example, we could completely overwrite the `insert` method in our piece type by including our own `insert` function in the `methods` section.
+:::
+
 ### `methods(self)`
+
+Add methods that can be invoked on `self` or from another module on `self.apos.modules['module-name']` or the designated module alias. Returns an object of functions.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  methods(self) {
+    return {
+      async averagePrice(req) {
+        let sum = 0;
+        const products = await self.find(req).toArray();
+
+        if (!products.length) {
+          return 0;
+        }
+
+        for (const product of products) {
+          sum += product.price;
+        }
+
+        return sum / products.length;
+      }
+    };
+  }
+};
+```
+
 #### `extendMethods(self)`
+
+<!-- TODO: Link to module reference section to find existing methods. -->
+Add to the functionality of a method inherited from the base class. This must return an object of functions, similar to [`methods`](#methods-self).
+
+Methods included should take a `_super` argument, followed by the normal arguments of the method being extended. If the original method took only a `req` argument, the extending method should take the arguments `_super, req`.
+
+To maintain the same application, they should return the same type of response as the original method. If the original returned an array of docs, the extension method should return an array of docs.
+
+```javascript
+// modules/product/index.js
+module.exports = {
+  // ...
+  extendMethods(self) {
+    return {
+      generate(_super, index) {
+        // Using _super with the original argument to generate a sample piece.
+        const piece = _super(index);
+        // Adding additional functionality.
+        piece.price = Math.random() * 100;
+        // Returning the generated piece, exactly as the original `generate`
+        // method does.
+        return piece;
+      }
+    };
+  }
+};
+```
+
 ### `components(self)`
+
+This function section returns an object containing functions that power asynchronous template components. These template components allow for asynchronous data requests in normally synchronous template rendering.
```

Comment:
"This function section" doesn't seem necessary. "These template components" -> "asynchronous template component".